L1Trigger/L1THGCal/plugins/HGCalTriggerDigiFEReproducer.cc

```diff
@@ -0,0 +1,117 @@
+    backEndProcessor_(conf.getParameterSet("BEConfiguration")) 
+/*****************************************************************/
+{
+    //setup geometry configuration
+    const edm::ParameterSet& geometryConfig = conf.getParameterSet("TriggerGeometry");
+    const std::string& trigGeomName = geometryConfig.getParameter<std::string>("TriggerGeometryName");
+    HGCalTriggerGeometryBase* geometry = HGCalTriggerGeometryFactory::get()->create(trigGeomName,geometryConfig);
+    triggerGeometry_.reset(geometry);
+
+    //setup FE codec
+    const edm::ParameterSet& feCodecConfig =  conf.getParameterSet("FECodec");
+    const std::string& feCodecName = feCodecConfig.getParameter<std::string>("CodecName");
+    HGCalTriggerFECodecBase* codec = HGCalTriggerFECodecFactory::get()->create(feCodecName,feCodecConfig);
+    codec_.reset(codec);
+    codec_->unSetDataPayload();
+
+    produces<l1t::HGCFETriggerDigiCollection>();
+    // register backend processor products
+    backEndProcessor_.setProduces(*this);
+}
+
+/*****************************************************************/
+void HGCalTriggerDigiFEReproducer::beginRun(const edm::Run& /*run*/, const edm::EventSetup& es) 
+/*****************************************************************/
+{
+    triggerGeometry_->reset();
+    HGCalTriggerGeometryBase::es_info info;
+    const std::string& ee_sd_name = triggerGeometry_->eeSDName();
+    const std::string& fh_sd_name = triggerGeometry_->fhSDName();
+    const std::string& bh_sd_name = triggerGeometry_->bhSDName();
+    es.get<IdealGeometryRecord>().get(ee_sd_name,info.geom_ee);
+    es.get<IdealGeometryRecord>().get(fh_sd_name,info.geom_fh);
+    es.get<IdealGeometryRecord>().get(bh_sd_name,info.geom_bh);
+    es.get<IdealGeometryRecord>().get(ee_sd_name,info.topo_ee);
+    es.get<IdealGeometryRecord>().get(fh_sd_name,info.topo_fh);
+    es.get<IdealGeometryRecord>().get(bh_sd_name,info.topo_bh);
+    triggerGeometry_->initialize(info);
+}
+
+/*****************************************************************/
+void HGCalTriggerDigiFEReproducer::produce(edm::Event& e, const edm::EventSetup& es)
+/*****************************************************************/
+{
+    std::unique_ptr<l1t::HGCFETriggerDigiCollection> fe_output( new l1t::HGCFETriggerDigiCollection );
+
+    edm::Handle<l1t::HGCFETriggerDigiCollection> digis_h;
+
+    e.getByToken(inputdigi_,digis_h);
+
+    const l1t::HGCFETriggerDigiCollection& digis = *digis_h;
+
+    for( const auto& digi_in : digis ) 
+    {    
+        const auto& module = triggerGeometry_->modules().at(digi_in.id());
+        fe_output->push_back(l1t::HGCFETriggerDigi());
+        l1t::HGCFETriggerDigi& digi_out = fe_output->back();
+        codec_->setDataPayload(*module, digi_in);
+        codec_->encode(digi_out);
+        digi_out.setDetId( digi_in.getDetId<HGCalDetId>() );
+        std::stringstream output;
```

Comment:
move the declaration of stringstream out of the loop and just clear it instead
@jbsauvan